src/analysis/inner_product_space/basic.lean

```diff
@@ -1801,6 +1805,11 @@ instance submodule.inner_product_space (W : submodule 𝕜 E) : inner_product_sp
 /-- The inner product on submodules is the same as on the ambient space. -/
 @[simp] lemma submodule.coe_inner (W : submodule 𝕜 E) (x y : W) : ⟪x, y⟫ = ⟪(x:E), ↑y⟫ := rfl
 
+lemma orthonormal_span {ι : Type*} {v : ι → E} (hv : orthonormal 𝕜 v) :
```

Comment:
Could you add a first lemma saying that if an orthonormal set of vectors is contained in a submodule `p`, then their `set.cod_restrict` is orthonormal in `p`?  Then deduce this from that.